samples/sample_common/src/brc_routines.cpp

```diff
@@ -458,11 +454,12 @@ mfxU16 GetFrameType(mfxU16 m_frameType, mfxU16 level, mfxU16 gopRegDist)
 
 
 bool  isFrameBeforeIntra (mfxU32 order, mfxU32 intraOrder, mfxU32 gopPicSize, mfxU32 gopRefDist)
- {
-     mfxI32 distance0 = gopPicSize*3/4;
-     mfxI32 distance1 = gopPicSize - gopRefDist*3;
-     return (order - intraOrder) > (mfxU32)(IPP_MAX(distance0, distance1));
- }
+{
+    mfxI32 distance0 = gopPicSize*3/4;
+    mfxI32 distance1 = gopPicSize - gopRefDist*3;
+    return mfxI32(order - intraOrder) > std::max(distance0, distance1);
```

Comment:
signed -> unsigned suspicious... Is that what we want to program?

Reply:
In general yes, looks like we did the thing we wanted. I checked this code with @takulako
```order - intraOrder >= 0``` according to code logic
```std::max(distance0, distance1) >= 0``` because ```distance0``` always ```>= 0```
So previous variant avoided signed/unsigned comparison mismatch by casting to unsigned, while mine is casting to signed. Should be the same